src/library/ResourceUtils/ResourceUtils.ts

```diff
@@ -84,7 +84,10 @@ export class ResourceUtils {
             const subPathValue = resourcePathValue[subPathIndex];
             if (this.isPrimitive(subPathValue)) {
               resultSet.push(subPathValue);
-            } else {
+            } else if (pathSections.length == 2) {
```

Comment:
== 2 because this means we only have the resource part of the path e.g. "Patient" and then the object we want to get e.g. "name". Before this only an empty array was returned even if names existed

Reply:
suggested a small alternative.